### ISSUE TITLE
[SUGGESTION] Add option to enable browser fingerprinting defenses 

### ISSUE THREAD

**Issue by ghost** [gorhill/uBlock#3125]
### Describe the issue

Tom added an option to enable browser fingerprinting defenses by setting `privacy.resistFingerprinting = true` https://bugzilla.mozilla.org/show_bug.cgi?id=1397611 with the goal that "privacy tools can toggle it on and take advantage of the work we've been doing."

This has been added for Firefox 58 so it's not until that time that one can take advantage of this.

Also see his patch: https://reviewboard.mozilla.org/r/177158/diff/2#index_header

**Comment by Thorin-Oakenpants** [comment]
Personally, I believe this is best left until it is more mature (at the very least). It will get an option in the preferences UI anyway, hopefully with a "Learn more" link. The target AFAIK is for 59 (since the next TBB will be based on ESR59).

If you add it, then due diligence means keeping up to date with the changes it causes and providing some info - and there is a LOT (including conflicts with existing prefs) ... and a LOT more to come. I would leave this in Mozilla's hands

For an idea of the scope of explaining this to people - see https://github.com/ghacksuserjs/ghacks-user.js/issues/7 .. for an idea of how this can conflict with existing prefs and alter your intended fingerprint, see https://github.com/ghacksuserjs/ghacks-user.js/issues/222

PS: looks like it will change from boolean to integer, as will privacy.resistFingerprinting = where 0=off, 1=on(global) and 2=(pb mode windows only) - see [1404017](https://bugzilla.mozilla.org/show_bug.cgi?id=1404017) and [1397624](https://bugzilla.mozilla.org/show_bug.cgi?id=1397624) .. so be aware of that

**Comment by gorhill** [comment]
Duplicate of #2381.
